src/lib/component/logger.hpp

```diff
@@ -99,9 +103,9 @@ namespace details {
         return make_stream(std::string(s), t);
     }
     //! @brief Makes a stream reference from a stream pointer.
-    template <typename S, typename T>
-    std::shared_ptr<std::ostream> make_stream(std::ostream* o, const common::tagged_tuple<S,T>&) {
-        return std::shared_ptr<std::ostream>(o, [] (void*) {});
+    template <typename O, typename S, typename T>
```

Comment:
You need a `if_ostream` guard here as well. Without it, the generic `make_stream` collides with that taking a string (also, all tests are passing? if so we definitely need to add some tests).